dotCMS/src/main/java/com/liferay/util/FileUtil.java

```diff
@@ -196,16 +197,26 @@ public static void copyFile(File source, File destination, boolean hardLinks) th
                 // setting this means we will try again if we cannot hard link
                 if (!destination.exists() || destination.length() == 0) {
                     hardLinks = false;
-                    Logger.warn(FileUtil.class, "Can't create hardLink. source: " + source.getAbsolutePath()
-                            + ", destination: " + destination.getAbsolutePath());
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Can't create hardLink. source: " + source.getAbsolutePath());
+                    sb.append(", destination: " + destination.getAbsolutePath());
+                    Logger.warn(FileUtil.class, sb.toString());
                 }
-            } catch (IOException e) {
-
-                hardLinks = false; // setting to false will execute the fallback
-                Logger.debug(FileUtil.class,
-                        "Could not created the hard link, will try copy for source: " + source +
-                        ", destination: " + destination + ". Error message: " + e.getMessage());
-            }
+            }  catch (Exception e1) {
+                if (e1 instanceof FileAlreadyExistsException){

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)